### PR TITLE
Add GH actions to push images to a DockerHub registry after a merge

### PR DESCRIPTION
**Pull request by NickLoukas**
Refs kubeflow/kubeflow#6502

### CODE REVIEW

**Review by kimwnasptd — COMMENTED** [review]
Thanks for the PR @NickLoukas! I've left some review comments.

The one regarding the image name holds for all workflows, but I only added it once.

File: .github/workflows/centraldb_docker_publish.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Login to DockerHub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Run TWA build and push
+      run: |
+        cd components/centraldashboard
+        export IMG=docker.io/kubeflownotebookswg
```
[the remark this answers is not in the document]
Comment (NickLoukas) [reply .github/workflows/centraldb_docker_publish.yaml]: Yes you are correct. I added the full name of the image in all the actions.

File: .github/workflows/jwa_docker_publish.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Build & Publish JWA Docker image
+on:
+  push:
+    branches:
+      - master
+      - v*-branch
+    paths:
+      - components/crud-web-apps/jupyter/**
```
[the remark this answers is not in the document]
Comment (NickLoukas) [reply .github/workflows/jwa_docker_publish.yaml:8]: Added that.

File: .github/workflows/centraldb_docker_publish.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Run TWA build and push
```
[the remark this answers is not in the document]
Comment (NickLoukas) [reply .github/workflows/centraldb_docker_publish.yaml]: Did it.

File: .github/workflows/kfam_docker_publish.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Run TWA build and push
```
[the remark this answers is not in the document]
Comment (NickLoukas) [reply .github/workflows/kfam_docker_publish.yaml]: Did it.

File: .github/workflows/nb_controller_docker_publish.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Run TWA build and push
```
[the remark this answers is not in the document]
Comment (NickLoukas) [reply .github/workflows/nb_controller_docker_publish.yaml]: Did it.

File: .github/workflows/vwa_docker_publish.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Run TWA build and push
```
[the remark this answers is not in the document]
Comment (NickLoukas) [reply .github/workflows/vwa_docker_publish.yaml:25]: Did it.

File: .github/workflows/nb_controller_docker_publish.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+name: Build & Publish Notebook Controller Docker image
+on:
+  push:
+    branches:
+      - master
+      - v*-branch
+    paths:
+      - components/notebook-controller/**
```
[the remark this answers is not in the document]
Comment (NickLoukas) [reply .github/workflows/nb_controller_docker_publish.yaml:8]: Added that.

**Comment by kimwnasptd** [comment]
@NickLoukas we just coordinated with @zijianjoy and added the credentials for DockerHub.

Can you
1. Use the value of the username directly and not from a secret, it's `kubeflownotebookswg`
2. Use the name `KUBEFLOWNOTEBOOKSWG_DOCKER_TOKEN` for the secret for the password

We used a token from DockerHub, which we should be able to use as a password https://github.com/marketplace/actions/docker-login

**Comment by kimwnasptd** [comment]
Thank you @NickLoukas! Let's unblock the repo

/lgtm
/approve

**Pull request merged**